core/src/main/java/org/elasticsearch/Build.java

```diff
@@ -79,10 +79,10 @@ static URL getElasticsearchCodebase() {
         return Build.class.getProtectionDomain().getCodeSource().getLocation();
     }
 
-    private String shortHash;
-    private String date;
+    private final String shortHash;
+    private final String date;
 
-    Build(String shortHash, String date, boolean isSnapshot) {
+    public Build(String shortHash, String date, boolean isSnapshot) {
```

Comment:
The three lines that contain the Build information are part of the larger `version` json object, so parsing them from xContent is a bit difficult. Also, it would still be good to be able to create random Build instances from the test.